### PR TITLE
new options parsed to StaticRouters, featuring stacking layouts like …

### PR DESCRIPTION
**Pull request by shpaw415**
nextjs like layouts stacking (now testing)

### CODE REVIEW

File: index.tsx [hunk 1]
```diff
@@ -89,32 +100,66 @@ export class StaticRouters {
...
+    let layoutsJsxList: React.ComponentType<{
+      children: React.ReactElement;
+    }>[] = [];
+    let index = 0;
+    for await (const i of layouts) {
+      const path = layouts.slice(0, index).join("/");
+      const pathToFile = `${this.baseDir}/${this.pageDir}${path}${this.options.layoutName}`;
+      console.log(path, index);
+      if (!(await Bun.file(pathToFile).exists())) continue;
+      const defaultExport = (await require(pathToFile)).default;
```
Comment (codehz) [inline index.tsx]: shouldn't be import? (yes I know require works in esm mode, but it is not async, and may not working with top level await)

File: index.tsx [hunk 2]
```diff
@@ -89,32 +100,66 @@ export class StaticRouters {
...
+      defaultExport && layoutsJsxList.push(defaultExport);
```
Comment (codehz) [inline index.tsx]: please use plain `if (defaultExport)` instead of &&

**Comment by codehz** [comment]
I think the stacking layout in nextjs is strictly depends on the App Router and Server Components, without those features, the stacking layout mode can cause many problems with the router...

But Server Components is hard to implement in bun currently... (without re-implementing the whole build system

**Comment by shpaw415** [comment]
now you can look at the final code and if you see something going wrong just tell me i modified the exemple to fit the new "use client"; directive at the start of an server componants

File: index.tsx (hunk 1 above)
Comment (shpaw415) [reply index.tsx]: i cannot say why but import could not be resolve for some reason

Comment (shpaw415) [reply index.tsx]: now import

**Comment by shpaw415** [comment]
for now the router for redirection do not work with the new system i will investigate on that if you do not find it first

**Comment by shpaw415** [comment]
i will soon push an update with ServerAction as well now working in beta version but much thing have changed for the build and router. i will do a new Exemple folder for that.

**Comment by shpaw415** [comment]
Link is now working. External and internal import are working

**Comment by shpaw415** [comment]
For now the imports of external files is complicated some edge case require more attention. when an export from an other file is made like ' export { someMethod } from "./file"; ' or '  ' export * from "./file" ' the bundler is acting weird and for now i dont know how to solve this...

**Comment by shpaw415** [comment]
if you consider adding to this project you can test with ( https://github.com/shpaw415/bunext ) i work on a framwork that is nextjs like and fully compatible with bun + some features

**Pull request closed without merging**